src/Symfony/Bundle/FrameworkBundle/Resources/config/messenger.xml

```diff
@@ -43,13 +43,12 @@
         </service>
 
         <!-- Message encoding/decoding -->
-        <service id="messenger.transport.serialize_message_with_type_in_headers" class="Symfony\Component\Messenger\Transport\Serialization\Serializer">
+        <service id="messenger.transport.serializer" class="Symfony\Component\Messenger\Transport\Serialization\Serializer">
             <argument type="service" id="serializer" />
+            <argument /> <!-- Format -->
+            <argument type="collection" /> <!-- Context -->
         </service>
 
-        <service id="messenger.transport.default_encoder" alias="messenger.transport.serialize_message_with_type_in_headers" public="true" />
-        <service id="messenger.transport.default_decoder" alias="messenger.transport.serialize_message_with_type_in_headers" public="true" />
```

Comment:
Is there a reason to move to PHP? If not, better keep here, isn't it?

Reply:
There is a reason: it's now configurable via the `framework.messenger.transport` configuration.